cmd/admin-handlers.go

```diff
@@ -1054,8 +1059,17 @@ func (a adminAPIHandlers) ObjectSpeedtestHandler(w http.ResponseWriter, r *http.
 		autotune = false
 	}
 
+	err = objectAPI.MakeBucketWithLocation(ctx, bucket, BucketOptions{})
+	if err != nil {
+		if _, ok := err.(BucketExists); !ok {
+			// Only BucketExists error can be ignored.
+			writeErrorResponseJSON(ctx, w, toAPIError(ctx, err), r.URL)
+			return
+		}
+	}
+
 	deleteBucket := func() {
-		objectAPI.DeleteBucket(context.Background(), pathJoin(minioMetaBucket, "speedtest"), DeleteBucketOptions{
+		objectAPI.DeleteBucket(context.Background(), bucket, DeleteBucketOptions{
```

Comment:
Let's not delete an existing bucket @krishnasrinivas or the one passed by client. For example if client passed a bucket name and MakeBucket failed with bucket exists.

Reply:
Also additionally I think we should write objects at a specific random prefix to avoid deleting existing content recursively if it's run on an existing bucket.

Reply:
> Also additionally I think we should write objects at a specific random prefix to avoid deleting existing content recursively if it's run on an existing bucket.

Yes ... correct. I think it will cause confusion (and risky) to take bucket name from the user as the user might affect the bucket's directory tree by mistake. So, for now I have made the change so that speedtest is done only on `minio-perf-test-tmp-bucket`. This way we don't have to worry about writing at certain prefix and how to purge things.

Currently if we want to test on a versioned (or encrypted) bucket we can create bucket `minio-perf-test-tmp-bucket` and enable features on this bucket and then do the speedtest.

If we see the need to take bucket name from user, we can make the change in future.